keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1266 @@
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+<!--
+Describe them, providing:
+  - API call type (e.g. PATCH pods)
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+Focusing mostly on:
+  - components listing and/or watching resources they didn't before
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+-->
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+<!--
+Describe them, providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+-->
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+<!--
+Describe them, providing:
+  - Which API(s):
+  - Estimated increase:
+-->
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+  Yes, but only when the `.spec.runPolicy.backoffLimitPerIndex` field is set.
+
+  - API type(s): Job
+  - Estimated increase in size:
+    - New `.status.failedIndexes` field in Status. In the worst-case scenario is
+      when every second index is failed, 10^4 indexes failed, and the last is
+      at completions-1; assuming completions are at 10^6 it translates to 70KB.
+      (see also [The Job object too big](#the-job-object-too-big)),
+    - New `.spec.runPolicy.backoffLimitPerIndex` field of `*int32` is 12 bytes.
+
+  - API type(s): Pod
+  - Estimated increase in size:
+    the new annotation `batch.kubernetes.io/job-index-failure-count` to keep the
+    current number of retries per index. Is around 50 bytes.
+
+
+<!--
+Describe them, providing:
+  - API type(s):
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+-->
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
```

Comment:
How about the dependency between steps 3 and 4 in tracking failed indexes?

Reply:
This may indeed increase the total number of pods for a short while as the finalizer removal is delayed. Added a note. Not sure though if there is any SLO/SLI covering this.